### PR TITLE
Bonnie 1628

### PR DESCRIPTION
**Pull request by mayerm94**
Data criteria types that do not use end_time in any way but bonnie still sends them over an end_time, they are then stored with an end_time which makes the excel exports of that datatype include a STOP time. Removing the end_times from these datacriteria makes the data types more closely align with the model info file.

Pull requests into CQL QDM Patient API require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1628
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases (It has been discussed and the one test that proves the functionality of the delete lines is sufficient given the amount of work on our plate)
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Code coverage has not gone down and all code touched or added is covered.
    * In rare situations, this may not be possible or applicable to a PR. In those situations:
        1. Note why this could not be done or is not applicable here:
        2. Add TODOs in the code noting that it requires a test
        3. Add a JIRA task to add the test and link it here:
- [x] Automated regression test(s) pass

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA test links: NA
- [x] Justification for using JIRA tests: NA
- [x] JIRA tests have been added to sprint NA

**Reviewer 1:**

Name: @hossenlopp
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name: @losborne
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

File: app/assets/javascripts/datatypes/datatype.js.coffee [hunk 1]
```diff
@@ -72,8 +72,12 @@ class CQL_QDM.QDMDatatype
     # Grab start and end time, format for proper display
     startTime = if @entry?.start_time then "START: #{moment.utc(@entry.start_time, 'X').format('MM/DD/YYYY h:mm A')}\n" else ""
     endTime = if @entry?.end_time then "STOP: #{moment.utc(@entry.end_time, 'X').format('MM/DD/YYYY h:mm A')}\n" else ""
-    # Get code if this datatype has one
-    code = @code() if @_codes
+     # If it is a patient characteristic, use getCode() instead of code()
```
Comment (losborne) [inline app/assets/javascripts/datatypes/datatype.js.coffee]: This is going to conflict with https://github.com/projecttacoma/cql_qdm_patientapi/pull/140/commits/6f64980424e1a70b1e904528527c8cb200fcf8cd

File: app/assets/javascripts/datatypes/assessment.js.coffee [hunk 2]
```diff
@@ -25,6 +25,7 @@ class CQL_QDM.AssessmentPerformed extends CQL_QDM.QDMDatatype
       @_result = @entry.values?[0]
     @_relatedTo = @entry.references
     @_components = @entry.components
+    delete @entry.end_time
```
Comment (hossenlopp) [inline app/assets/javascripts/datatypes/assessment.js.coffee]: The patient api shouldn't be modifying the HDS model. It should only adapt it differently. I don't understand why it is deleting the end_time instead of simply not exposing on the QDM side.

Comment (mayerm94) [reply app/assets/javascripts/datatypes/assessment.js.coffee]: (Posting for visibility) Bonnie adds end_time to all data-criteria. Deleting it off of entry is required so that the toString on the dataElement base type does not include an end_date for types that it is not relevant to
